### PR TITLE
etcdserver: add maintain service to support defrag

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony

### CODE REVIEW

File: etcdserver/api/v3rpc/maintenance.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+func NewMaintenanceServer(s *etcdserver.EtcdServer) pb.MaintenanceServer {
+	return &maintenanceServer{bg: s}
+}
+
+func (ms *maintenanceServer) Defragment(ctx context.Context, sr *pb.DefragmentRequest) (*pb.DefragmentResponse, error) {
+	plog.Noticef("starting to defragment the storage backend...")
+	err := ms.bg.Backend().Defrag()
+	if err != nil {
+		return nil, err
```
Comment (heyitsanthony) [inline etcdserver/api/v3rpc/maintenance.go]: `Noticef` that it failed?

Comment (xiang90) [inline etcdserver/api/v3rpc/maintenance.go]: probably errorf?

Comment (heyitsanthony) [inline etcdserver/api/v3rpc/maintenance.go]: of course

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**